angel_system/eval/visualization.py

```diff
@@ -151,3 +155,78 @@ def plot_pr_curve(self, detect_intersection_thr=0.1):
         ax.set_ylabel("Precision")
 
         fig.savefig(f"{self.output_dir}/PR.png")
+
+    def plot_roc_curve(self, detect_intersection_thr=0.1):
+        # ============================
+        # Setup figure
+        # ============================
+        fig, ax = plt.subplots(figsize=(7, 8))
+
+        ax.set_xlim([0.0, 1.0])
+        ax.set_ylim([0.0, 1.05])
+        ax.set_title("Receiver Operating Characteristic (ROC)")
+        ax.set_xlabel("False Positive Rate")
+        ax.set_ylabel("True Positive Rate")
+
+        colors = plt.cm.rainbow(np.linspace(0, 1, len(self.labels)))
+
+        # ============================
+        # Get ROC and AUC per class 
+        # ============================
+        fpr = dict()
+        tpr = dict()
+        roc_auc = dict()
+        for i, row in self.labels.iterrows():
```

Comment:
Also intersecting the PR curve output a little here: This global for-loop is adding all labels to a single plot. This will lead to overcrowding as we get more and more class labels into the plot. We may want to have an option to, for both PR and ROC curves, output single-class plots out to independent files, or express a limit to the number of curves on a single plot. E.g. if there are 30 labels, limit only 5 at a time being rendered on a plot, thus creating 6 files.

Reply:
I added subfolders for roc and pr and switched it so it writes out individual label plots in addition to the overall plot.